neqo-transport/src/send_stream.rs

```diff
@@ -792,21 +816,23 @@ impl SendStream {
                     } else {
                         Some((offset, slice))
                     }
-                })
+                } else {
+                    None
+                }
             }
             SendStreamState::DataSent {
-                ref send_buf,
+                ref mut send_buf,
                 fin_sent,
                 ..
             } => {
-                let bytes = send_buf.next_bytes();
-                if bytes.is_some() {
-                    bytes
+                let used = send_buf.used(); // immutable first
```

Comment:
I think that you just want to  add this line, not change the next two.  Your change might involve an allocation (though the compiler is probably smart enough to avoid that, no point in writing `Some((offset, slice))` twice.

Reply:
The compiler suggested I make this change (failed 2 checks, so I changed it to make the compiler happy via 'more cleanup').  I'll try changing it back...

Reply:
And this time it passes.  Dunno, ok

Reply:
Yeah, it's funny, but I think that you can also move the call to used back.  The only change being to the addition of 'mut'.